service/paxos/paxos_state.cc

```diff
@@ -366,11 +368,104 @@ static sstring paxos_state_cf_filter(const schema& s, const schema& state_schema
         : sstring();
 }
 
-paxos_store::paxos_store(db::system_keyspace& sys_ks)
-: _sys_ks(sys_ks)
+static const sstring paxos_state_table_suffix = "_scylla_paxos_state";
+
+static sstring paxos_state_table_name(const schema& s) {
+    return sstring(s.cf_name()) + paxos_state_table_suffix;
+}
+
+static schema_ptr try_get_paxos_state_schema(const replica::database& db, const schema& s) {
+    // Throw if the base table has been dropped or recreated.
+    // We can't rely on s.table() since a table is stored in tables_metadata as an lw_shared_ptr,
+    // and can live for a while without being referenced from it.
+    db.find_column_family(s.id());
+
+    const auto& tables = db.get_tables_metadata();
+    const auto state_table_id = tables.get_table_id_if_exists({s.ks_name(), paxos_state_table_name(s)});
+    return state_table_id ? tables.get_table(state_table_id).schema() : nullptr;
+}
+
+static schema_ptr create_paxos_state_schema(const schema& s, std::optional<table_id> table_id) {
+    schema_builder builder(table_id, s.ks_name(), paxos_state_table_name(s),
+        // partition key
+        {{"row_key", bytes_type}}, // byte representation of a row key that hashes to the same token as original
+        // clustering key
+        {},
+        // regular columns
+        {
+            {"promise", timeuuid_type},
+            {"most_recent_commit", bytes_type}, // serialization format is defined by frozen_mutation idl
+            {"most_recent_commit_at", timeuuid_type},
+            {"proposal", bytes_type}, // serialization format is defined by frozen_mutation idl
+            {"proposal_ballot", timeuuid_type},
+        },
+        // static columns
+        {},
+        // regular column name type
+        utf8_type,
+        // comment
+        fmt::format("Paxos state for {}.{}", s.ks_name(), s.cf_name())
+    );
+    builder.set_gc_grace_seconds(0);
+    builder.with_hash_version();
+    return builder.build(schema_builder::compact_storage::no);
```

Comment:
But, this table should not be repaired or replicated, right?

Reply:
It's never replicated. It may be repaired in case of base table rebuild or repair.

Reply:
Need to prevent read-repair etc.

I guess it's not replicated because you don't use coordinator APIs. I think it's okay.

Reply:
Makes sense. Filtering is suboptimal compared to preparing a replica set of locals in advance (in effective_replication_map) but we can optimize later.